### ISSUE TITLE
GosuCompile shouldn't rely on SourceTask.source field

### ISSUE THREAD

**Issue by lptr** [gosu-lang/gradle-gosu-plugin#47]
In `getSourceRoots()` there is this code that refers to the `source` field from `SourceTask`:

```java
    for(Object obj : this.source) {
      if(obj instanceof SourceDirectorySet) {
        returnValues.addAll(((SourceDirectorySet) obj).getSrcDirs());
      }
    }
```

The `source` field was exposed before as `protected`, but we are removing that in Gradle 5.0. Please use `getSource()` instead.

**Comment by DPUkyle** [comment]
At first glance, I thought it was just sloppy programming. But there is a reason I originally accessed the field directly - I need the raw values, some of which are instances of SourceDirectorySet.

It's ugly but for now I'll have to resort to reflection 👎

**Comment by lptr** [comment]
Why do you need the `SourceDirectorySet`? Why does gosu need both the source directories and the sources themselves? Could you maybe calculate the source directories by observing the source files (and collecting common root dirs)?

**Comment by DPUkyle** [comment]
Yes that can work. There are some complicating factors though; I'll be in touch via Slack.

**Comment by DPUkyle** [comment]
See https://github.com/gradle/gradle/pull/6845/files#diff-d735159ef34e50bb8dcb7c66723e898dR238
